src/generic_requirement_type.h

```diff
@@ -0,0 +1,85 @@
+#pragma once
+#ifndef CATA_SRC_GENERIC_REQUIREMENT_TYPE_H
+#define CATA_SRC_GENERIC_REQUIREMENT_TYPE_H
+
+#include <string>
+#include <climits>
+
+#include "bodypart.h"
+#include "field.h"
+#include "translations.h"
+#include "type_id.h"
+#include "weather_type.h"
+
+template<typename T>
+class generic_factory;
+class Character;
+
+struct w_point;
+
+enum class time_requirement_type : int {
+    day,
+    night,
+    both,
+    last
+};
+template<>
+struct enum_traits<time_requirement_type> {
+    static constexpr time_requirement_type last = time_requirement_type::last;
+};
+
+struct generic_requirement_type {
+    public:
+        friend class generic_factory<generic_requirement_type>;
+        bool was_loaded = false;
+        generic_requirement_type_id id;
+        int windpower_min = INT_MIN;
```

Comment:
As in `generic_event_type`, I think a better design is to have a vector of requirements implemented via an abstract base class, so that `test` doesn't need to check all of the things that aren't relevant to a particular requirement.